src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -149,9 +183,70 @@ public ReduceableSearchResult reduce(Collection<Collector> collectors) {
             TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(newTopDocs, hybridTopScoreDocCollector.getMaxScore());
             return (QuerySearchResult result) -> { result.topDocs(topDocsAndMaxScore, getSortValueFormats(sortAndFormats)); };
         }
+
+        if (!hybridSortedTopDocCollectors.isEmpty()) {
+            HybridTopFieldDocSortCollector hybridSortedTopScoreDocCollector = hybridSortedTopDocCollectors.stream()
```

Comment:
please add a TODO note here, we already know that this logic is incorrect for case of multiple segment slices -> multiple collectors.